File: Sample/iOS_OSX/Sample-iOS/ModuleViewControllers/ThreadModuleViewController.m
//
//  ThreadModuleViewController.m
//  Sample
//
//  Created by 冯鸿杰 on 2019/1/1.
//  Copyright © 2019年 vimfung. All rights reserved.
//

#import "ThreadModuleViewController.h"
#import "LuaScriptCore.h"

static LSCScriptController *config = nil;

@interface ThreadModuleViewController ()

@property (nonatomic, strong) LSCContext *context;

@end

@implementation ThreadModuleViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    config = [[LSCScriptController alloc] init];
    
    self.context = [[LSCContext alloc] init];
    [self.context onException:^(NSString *message) {
        
        NSLog(@"lsc exception = %@", message);
        
    }];
    
    [self.context evalScriptFromFile:@"Thread-Sample.lua"];
}

#pragma mark - Table view data source

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    switch (indexPath.row)
    {
        case 0:
            //Run Thread
            [self.context evalScriptFromString:@"Thread_Sample_run()"];
            break;
        case 1:
            //Stop Thread
            [self.context evalScriptFromString:@"Thread_Sample_stop()"];
            break;
        default:
            break;
    }
}

@end

File: Sample/iOS_OSX/Sample-iOS-Swift/ModuleViewControllers/ThreadModuleViewController.swift
//
//  ThreadModuleViewController.swift
//  Sample-iOS-Swift
//
//  Created by 冯鸿杰 on 2019/1/2.
//  Copyright © 2019年 vimfung. All rights reserved.
//

import UIKit
import LuaScriptCore_iOS_Swift

class ThreadModuleViewController: UITableViewController {

    let context : LuaContext = LuaContext();
    
    override func viewDidLoad() {
        super.viewDidLoad()

        context.onException { (msg) in
            print("lua exception = \(msg!)");
        }
        
        _ = context.evalScript(filePath: "Thread-Sample.lua");
    }

    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        switch indexPath.row
        {
        case 0:
            _ = context.evalScript(script: "Thread_Sample_run()");
        default:
            break;
        }
    }

}
